test/test_learn_detect.py

```diff
@@ -41,18 +71,80 @@ def callback(data):
         rospy.logerr(e)
         return
 
-    cv2.imshow("Image window", cv_image)
-    key = cv2.waitKey(10)
+    if _SHOW:
+	cv2.imshow("Image window", cv_image)
+
+    if ( _INTERACTIVE ):
+        key = cv2.waitKey(10)
+    else:
+        cv2.waitKey(10)
+        key = 1048676
 
     if key == 1048684 or key == 108: # L
         print learn_srv(image=data, name=raw_input("Name? "))
     elif key == 1048676 or key == 100: # D
-        print detect_srv(image=data, external_api_request=external_api_request)
+        detections = detect_srv(image=data, external_api_request=external_api_request)
+        #print detections
+
+        if ( detections.face_detections.__len__() == 0):
+            disp_str = "(no detections)"
+            if ( camera_lr ):
```

Comment:
You don't need this if statement.  Log messages should be happening in the function calls below